### ISSUE TITLE
Content security policy blocking SimpleLogin username generation in browser - self hosted

### ISSUE THREAD

**Issue by vinnyperella** [bitwarden/clients#4179]
### Steps To Reproduce

1. Log in to Bitwarden self hosted web vault using web browser (tested in both Chromium and Firefox).
2. Generate API key from SimpleLogin dashboard.
3. Go to "Tools -> Generator" in Bitwarden web vault and attempt to generate a 'Forwarded Email Alias' username using SimpleLogin as the service and the API key generated in step 2.

### Expected Result

The username is generated successfully and shows up in the SimpleLogin dashboard.

### Actual Result

The page immediately errors with the following text in the upper right (red box) "NetworkError when attempting to fetch resource" and the following details in the browser console.

Firefox:
Content Security Policy: The page’s settings blocked the loading of a resource at https://app.simplelogin.io/api/alias/random/new (“connect-src”).

Chromium:
Refused to connect to 'https://app.simplelogin.io/api/alias/random/new' because it violates the following Content Security Policy directive: "connect-src 'self' wss://*************** https://api.pwnedpasswords.com https://twofactorauth.org".
Refused to connect to 'https://app.simplelogin.io/api/alias/random/new' because it violates the document's Content Security Policy.

### Additional Context

This only occurs in the web browser it does not occur in the Firefox browser add-on I am able to successfully create usernames using the same procedure in the browser add-on. I also tried searching the Github issues and couldn't find another issue that describes what I am seeing however I did find a Bitwarden community forum post with the exact same problem from back in June. Unfortunately there is no resolution or information confirming the bug/issue so I figured I should open a Github issue https://community.bitwarden.com/t/email-alias-generator/41785/12.

I believe the SimpleLogin URL (https://app.simplelogin.io/api/alias/random/new) needs added to the "nginx_header_content_security_policy" parameter as I see there's an entry for haveibeenpwned which is another API the web vault uses to check if your email was included in a hacked credential dump.

bitwarden@bitwarden:/opt/bitwarden/bwdata$ grep nginx_header_content_security_policy config.yml 
nginx_header_content_security_policy: "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data: https://haveibeenpwned.com https://www.gravatar.com; child-src 'self' https://*.duosecurity.com https://*.duofederal.com; frame-src 'self' https://*.duosecurity.com https://*.duofederal.com; connect-src 'self' wss://{0} https://api.pwnedpasswords.com https://twofactorauth.org; object-src 'self' blob:;"
bitwarden@bitwarden:/opt/bitwarden/bwdata$

### Operating System

Linux

### Operating System Version

Ubuntu 19.04

### Web Browser

Chrome, Firefox

### Browser Version

Firefox 109.0a1, Chromium 110.0.5427.0

### Build Version

Version 2022.10.0 (self hosted up-to-date)

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by vinnyperella** [comment]
I added https://app.simplelogin.io/api/alias/random/new to the "connect-src 'self'" policy for nginx parameter nginx_header_content_security_policy in "bwdata/config.yml" and rebuilt Bitwarden config (./bitwarden.sh rebuild) and it resolved the issue. I can now successfully create forward email aliases using SimpleLogin in the Bitwarden web vault (self hosted) username generator.

```
bitwarden@bitwarden:/opt/bitwarden/bwdata$ grep nginx_header_content_security_policy config.yml 
nginx_header_content_security_policy: "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data: https://haveibeenpwned.com https://www.gravatar.com; child-src 'self' https://*.duosecurity.com https://*.duofederal.com; frame-src 'self' https://*.duosecurity.com https://*.duofederal.com; connect-src 'self' wss://{0} https://api.pwnedpasswords.com https://twofactorauth.org https://app.simplelogin.io/api/alias/random/new; object-src 'self' blob:;"
bitwarden@bitwarden:/opt/bitwarden/bwdata$
```

**Comment by vinnyperella** [comment]
This is still an issue I'm kind of surprised this hasn't at least been looked in to I still cannot create aliases in SimpleLogin using the web vault.

**Comment by phillipmcmahon** [comment]
It's still an issue, but you can follow the steps above, add the necessary parts to bwdata/config.yml, rebuild and it works fine.

**Comment by vinnyperella** [comment]
> It's still an issue, but you can follow the steps above, add the necessary parts to bwdata/config.yml, rebuild and it works fine.
> 
> 

You are correct however this should be resolved on the Bitwarden side they implemented a feature that clearly wasn't tested which is probably the most frustrating aspect of this issue as I don't really intend to use the feature. A simple "we're aware and are looking in to it" would have sufficed.

**Comment by atjbramley** [comment]
@vinnyperella, 

Thank you for your report! 

I was able to reproduce this issue, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Apologise for the delay in response - this was missed from our regular checks.

Thanks once again!

**Comment by vinnyperella** [comment]
> @vinnyperella, 
> 
> Thank you for your report! 
> 
> I was able to reproduce this issue, and I have flagged this to our engineering team.
> 
> If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.
> 
> Apologise for the delay in response - this was missed from our regular checks.
> 
> Thanks once again!

I still don't believe this has been fixed I don't see the SimpleLogin URL in the content security policy header for self hosted instances.

**Comment by audreyality** [comment]
Hi @vinnyperella! 

Thanks for letting us know this was broken. It's been implemented in #self-host/316 and should be included in the 2024.12 release.
